app/utils/common.ts

```diff
@@ -120,3 +123,25 @@ export function joinList(
         },
     ).flat();
 }
+
+export function hasNoData(obj: unknown): boolean {
+    if (obj === undefined || obj === null || isNaN(obj)) {
+        return true;
+    }
+
+    if (isList(obj)) {
+        if (obj.length <= 0) {
+            return true;
+        }
+        return false;
+    }
+
+    if (isObject(obj)) {
+        if (Object.keys(obj).length <= 0) {
+            return true;
+        }
+        return false;
+    }
+
+    return false;
```

Comment:
We should also handle string.
Most of the time "" is also treated as empty filter.

We should also think about cases like these where it get's a bit fuzzy:
```
{
    startDate: '',
    endDate: '',
}
```